share/spice/detect_lang/detect_lang.js

```diff
@@ -108,42 +87,82 @@ function ddg_spice_detect_lang (api_result) {
 
         return langs[language] || "";
     };
-    Handlebars.registerHelper("DetectLang_expandLang", expandLang);
 
-    if(expandLang(api_result.data.detections[0].language) === "") {
-        return;
+    var formatTitle = function (first, second) {
+        var title = "This text is";
+        if (first.isReliable) {
+            title += " definitely ";
+        } else {
+            title += " probably ";
+        }
+        title += expandLang(first.language);
+        title += " (" + getPercentConfidence(first.confidence) + ").";
+        return title;
     }
 
-    var d0 = api_result.data.detections[0],
-        d1 = null;
-
-    if (api_result.data.detections.length > 1) {
-        d1 = api_result.data.detections[1];
+    var formatSubtitle = function (second) {
+        var subtitle = null;
+        if (!second) {
+            return subtitle;
+        }
+        subtitle = "But it could also be " + expandLang(second.language);
+        subtitle += " (" + getPercentConfidence(second.confidence) + ").";
+        return subtitle;
     }
 
-    // Display the plug-in.
-    Spice.add({
-        id: 'detect_lang',
-        data: { first: d0, second: d1 },
-        name: "Answer",
-        meta: {
-            sourceUrl       : "http://detectlanguage.com/",
-            sourceName      : "Detect Language",
-        },
-	signal: 'high',
-        templates: {
-            group: 'base',
-            options: {
-                content: Spice.detect_lang.content,
-		moreAt: true
+    env.ddg_spice_detect_lang = function (api_result) {
+
+        // Check for any errors.
+        if(!api_result || !api_result.data || !api_result.data.detections || api_result.data.detections.length === 0) {
+            return Spice.failed('detect_lang');
+        }
+
+        api_result.data.detections.sort(function(a, b) {
+            if(a.confidence > b.confidence) {
+                return -1;
+            } else if(a.confidence < b.confidence) {
+                return 1;
             }
+            return 0;
+        });
+
+        if(expandLang(api_result.data.detections[0].language) === "") {
+            return;
         }
-        
-    });
-};
 
-Handlebars.registerHelper("DetectLang_toPercent", function(confidence) {
-    "use strict";
-    var percentage = Math.round(confidence * 100);
-    return (percentage > 100 ? 100 : percentage) + "% sure";
-});
+        var d0 = api_result.data.detections[0],
+            d1 = null;
```

Comment:
Could `d0` and `d1` be renamed so that we get a bit more context on what those variables represent?